shopinvader/services/cart.py

```diff
@@ -440,17 +447,24 @@ def _get(self, create_if_not_found=True):
             return self._create_empty_cart()
         return cart
 
-    def _create_empty_cart(self):
-        vals = self._prepare_cart()
+    def _extract_cart_params(self, **params):
+        return self._extract_extra_params("cart__", **params)
+
+    def _create_empty_cart(self, **cart_params):
+        vals = self._prepare_cart(**cart_params)
         return self.env["sale.order"].create(vals)
 
-    def _prepare_cart(self):
+    def _prepare_cart(self, **cart_params):
         partner = self.partner or self.shopinvader_backend.anonymous_partner_id
-        vals = {
-            "typology": "cart",
-            "partner_id": partner.id,
-            "shopinvader_backend_id": self.shopinvader_backend.id,
-        }
+        vals = cart_params.copy()
```

Comment:
can you give example of what kind of parameters  you need here ?

Reply:
It can be any param. Eg: the customer must set a delivery date (commitment_date normally).

Reply:
but you still have to write a validator for allowing commitment_date ?!

Reply:
if the key is a new one: yes, of course.
But read my comment: I would include `_validator_update` here like this (did it in my custom module):

```py
        validator = super()._validator_add_item()
        cart_update_validator = {
            "cart__" + k: v for k, v in self._validator_update().items()
        }
        validator.update(cart_update_validator)
```
so that you rely on the same schema of validator_update w/out worrying about missing keys.